### ISSUE TITLE
Syntax highlighting for unique node names is not working with inline annotations

### ISSUE THREAD

**Issue by Kubulambula** [godotengine/godot#79755]
### Godot version

4.1.1

### System information

Any

### Issue description

When using unique names with annotations on one line, the unique name is not highlighted, but the script is still valid.

![capture](https://github.com/godotengine/godot/assets/72454967/201b4615-14aa-46fc-b20a-90721b048550)

### Steps to reproduce

Have an annotation and inline unique name like this:
`@warning_ignore("unsafe_property_access") %MyLabel.text = "Hello World!"`.

### Minimal reproduction project

[reproduction.zip](https://github.com/godotengine/godot/files/12131147/reproduction.zip)

**Comment by Ybaedbaenn** [comment]
I can confirm that the bug happens for me on android editor with:
###### Godot v4.1.1.stable • LineageOS 18.1 • 20230614-NIGHTLY • a5y17lte • Vulkan (Mobile) • integrated Mali-T830 • (4 Threads)

**Comment by dalexeev** [comment]
In this context, the `%`character is treated as a string remainder / formatting operator because it comes after the "operand" (close parenthesis). I don't think it's possible to fix this without a major refactoring of the highlighter. But I don't mean that this is a reason to close this issue, just clarifying that we probably shouldn't expect a fix soon.

To me, using inline annotations with statements looks like an undesirable style. This is ok with short definitions (like `@onready` and `@export`), but otherwise I prefer annotations on separate lines.

**Comment by Kubulambula** [comment]
> To me, using inline annotations with statements looks like an undesirable style. This is ok with short definitions (like `@onready` and `@export`), but otherwise I prefer annotations on separate lines.

I agree, but sometimes, you just have a lot of these lines and want to save space. That's how I discovered it.

**Comment by dalexeev** [comment]
> but sometimes, you just have a lot of these lines and want to save space

* In 3.x there is a `warning-ignore-all` comment. There is no analogue in 4.x yet, but there is a PR #76020.
